MyApplication/app/src/main/java/com/joelimyx/myapplication/MainActivity.java

```diff
@@ -0,0 +1,114 @@
+package com.joelimyx.myapplication;
+
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+import android.support.v4.net.ConnectivityManagerCompat;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.support.v7.view.menu.MenuAdapter;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.Button;
+import android.widget.Toast;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class MainActivity extends AppCompatActivity {
+    RecyclerView mRecyclerView;
+    @BindView(R.id.button1) Button mButton1;
+    Button mButton2;
+    @BindView(R.id.button3) Button mButton3;
+    public static final StringBuilder GET_URL_BUILDER = new StringBuilder("http://api.walmartlabs.com/v1/search?format=json&apiKey=xp8nz8badbdjrn2pb9r7bsub&query=");
+    private MainAdapter mAdapter;
```

Comment:
All of these can be `private`. You have one private, one public, and the rest are "package private" (the default when you don't enter an access modifier). Try to be consistent with these, and generally, you want to set the access to be as restrictive as possible to avoid unexpected changes to your variables.